model/src/ratio_as_decimal.rs

```diff
@@ -0,0 +1,137 @@
+use bigdecimal::{BigDecimal, One};
+use num::bigint::Sign as Sign04;
+use num::BigRational;
+use num_bigint::{BigInt, Sign as Sign03};
+use serde::Deserialize;
+use serde::{de, Deserializer, Serializer};
+use serde_with::{DeserializeAs, SerializeAs};
+use std::borrow::Cow;
+use std::str::FromStr;
+
+pub struct DecimalBigRational;
+
+impl<'de> DeserializeAs<'de, BigRational> for DecimalBigRational {
+    fn deserialize_as<D>(deserializer: D) -> Result<BigRational, D::Error>
+    where
+        D: Deserializer<'de>,
+    {
+        deserialize(deserializer)
+    }
+}
+
+impl<'de> SerializeAs<BigRational> for DecimalBigRational {
+    fn serialize_as<S>(source: &BigRational, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: Serializer,
+    {
+        serialize(source, serializer)
+    }
+}
+
+pub fn serialize<S>(value: &BigRational, serializer: S) -> Result<S::Ok, S::Error>
+where
+    S: Serializer,
+{
+    let top_bytes = value.numer().to_bytes_le();
+    let top = BigInt::from_bytes_le(sign_04_to_03(top_bytes.0), &top_bytes.1);
+    let bottom_bytes = value.denom().to_bytes_le();
+    let bottom = BigInt::from_bytes_le(sign_04_to_03(bottom_bytes.0), &bottom_bytes.1);
+    let decimal = BigDecimal::from(top) / BigDecimal::from(bottom);
+    serializer.serialize_str(&decimal.to_string())
+}
+
+pub fn deserialize<'de, D>(deserializer: D) -> Result<BigRational, D::Error>
+where
+    D: Deserializer<'de>,
+{
+    let value_decimal: Result<BigDecimal, D::Error> =
+        BigDecimal::from_str(&*Cow::<str>::deserialize(deserializer)?).map_err(|err| {
+            de::Error::custom(format!("failed to decode decimal BigDecimal: {}", err))
+        });
+
+    match value_decimal {
+        Ok(big_decimal) => {
+            let (x, exp) = big_decimal.into_bigint_and_exponent();
+            let numerator_bytes = x.to_bytes_le();
+            let base = num::bigint::BigInt::from_bytes_le(
+                sign_03_to_04(numerator_bytes.0),
+                &numerator_bytes.1,
+            );
+            let ten = BigRational::new(num::bigint::BigInt::from(10), num::bigint::BigInt::from(1));
```

Comment:
nit: this is semantically identical, but more succinct 
```suggestion
            let ten = BigRational::new(10.into(), 1.into());
```

Reply:
Also, if you want to be fancy, you can make this a `lazy_static`.

Reply:
I might just.